src/lib/homestore_backend/tests/test_shard_manager.cpp

```diff
@@ -176,223 +161,144 @@ TEST_F(ShardManagerWithShardsTesting, ListShards) {
-        EXPECT_TRUE(info.id == shard_info.id);
-        EXPECT_TRUE(info.placement_group == _pg_id);
-        EXPECT_EQ(info.state, ShardInfo::State::OPEN);
-    });
-
-    // everything is fine;
-    ho->on_pre_commit_shard_msg(100, sisl::blob(buf->data_begin(), needed_size), sisl::blob(), nullptr);
-    ho->on_shard_message_commit(100, sisl::blob(buf->data_begin(), needed_size), sisl::blob(), nullptr);
-    future = _home_object->shard_manager()->get_shard(shard_info.id).get();
-    ASSERT_TRUE(!!future);
-    future.then([this, shard_info](auto const& info) {
-        EXPECT_TRUE(info.id == shard_info.id);
-        EXPECT_TRUE(info.placement_group == _pg_id);
-        EXPECT_EQ(info.state, ShardInfo::State::SEALED);
-    });
+    homeobject::HSHomeObject* ho = dynamic_cast< homeobject::HSHomeObject* >(_home_object.get());
+    auto pg = dp_cast< homeobject::HSHomeObject::HS_PG >(ho->_pg_map[_pg_id]);
+    auto repl_dev = pg->repl_dev_;
+    const auto msg_size = sisl::round_up(seal_shard_msg.size(), repl_dev->get_blk_size());
+    auto req = homeobject::repl_result_ctx< ShardManager::Result< ShardInfo > >::make(msg_size, 512 /*alignment*/);
+    auto buf_ptr = req->hdr_buf_.bytes;
+    std::memset(buf_ptr, 0, msg_size);
+    std::memcpy(buf_ptr, seal_shard_msg.c_str(), seal_shard_msg.size());
+
+    req->header_.msg_type = homeobject::ReplicationMessageType::SEAL_SHARD_MSG;
+    req->header_.pg_id = _pg_id;
+    req->header_.shard_id = shard_info.id;
+    req->header_.payload_size = msg_size;
+    req->header_.payload_crc = crc32_ieee(homeobject::init_crc32, buf_ptr, msg_size);
+    req->header_.seal();
+    sisl::blob header;
+    header.bytes = r_cast< uint8_t* >(&req->header_);
+    header.size = sizeof(req->header_);
+    sisl::sg_list value;
+    value.size = msg_size;
+    value.iovs.push_back(iovec(buf_ptr, msg_size));
+    repl_dev->async_alloc_write(header, sisl::blob{}, value, req);
+    auto info = req->result().get();
+    EXPECT_TRUE(info);
+    EXPECT_TRUE(info.value().id == shard_info.id);
+    EXPECT_TRUE(info.value().placement_group == _pg_id);
+    EXPECT_EQ(info.value().state, ShardInfo::State::SEALED);
 
     auto pg_iter = ho->_pg_map.find(_pg_id);
     EXPECT_TRUE(pg_iter != ho->_pg_map.end());
-    auto& pg = pg_iter->second;
-    EXPECT_EQ(1, pg->shards.size());
-    auto& check_shard = *pg.shards.begin();
-    EXPECT_EQ(ShardInfo::State::SEALED, check_shard.info.state);
-    EXPECT_TRUE(check_shard.metablk_cookie != nullptr);
+    auto pg_result = pg_iter->second;
+    EXPECT_EQ(1, pg_result->shards_.size());
+    auto& check_shard = pg_result->shards_.front();
+    EXPECT_EQ(ShardInfo::State::SEALED, check_shard->info.state);
 }
 
-TEST_F(ShardManagerWithShardsTesting, ShardManagerRecovery) {
+class FixtureAppWithRecovery : public FixtureApp {
+public:
+    std::list< std::filesystem::path > devices() const override {
+        const std::string fpath{"/tmp/test_homestore.data"};
```

Comment:
This same file name will clash with other test binary (not test cases in this cpp), if they run in parallel, right?
Can we apply `/tmp/<test_binary_name>_homestore.data` to make sure every test binary use their unique file name?

Reply:
ok, I will align with Hari's PR to add a random number to file path